api/src/modules/h3-data/h3-data.service.ts

```diff
@@ -410,30 +408,46 @@ export class H3DataService {
       getImpactMapDto.supplierIds,
     );
 
+    const materialsH3DataYears: MaterialsH3DataYears[] = [];
+
+    if (getImpactMapDto.materialIds?.length)
+      for (const materialId of getImpactMapDto.materialIds) {
+        const material: Material = await this.materialService.getMaterialById(
+          materialId,
+        );
+        const materialHarvestH3DataYear: number | undefined =
+          await this.filterYearsByLayerService.getH3MaterialYearForCalculations(
+            materialId,
+            MATERIAL_TO_H3_TYPE.HARVEST,
+            getImpactMapDto.year,
+          );
+        const materialProducerH3DataYear: number | undefined =
+          await this.filterYearsByLayerService.getH3MaterialYearForCalculations(
+            materialId,
+            MATERIAL_TO_H3_TYPE.PRODUCER,
+            getImpactMapDto.year,
+          );
+        materialsH3DataYears.push(
+          {
+            materialName: material.name,
+            materialDataYear: materialHarvestH3DataYear,
+            materialDataType: MATERIAL_TO_H3_TYPE.HARVEST,
+          },
+          {
+            materialName: material.name,
+            materialDataYear: materialProducerH3DataYear,
+            materialDataType: MATERIAL_TO_H3_TYPE.PRODUCER,
+          },
```

Comment:
Are our beloved frontends aware that you are retrieving this info when sending a map back? I understand the reasoning of adding this info, but I'd like to know if this is going to be actually used, as any kind of map is already a huge payload and I think we should try to retrieve just the exact needed amount of data

Reply:
My idea was to include in the response all the relevant information and then present it to David and Elena for feedback. In terms of information, in my opinion, it is correct if we want to have full transparency on the whole years situation. Impact map can be requested for multiple materials, lets say user wants impact map for cocoa and cereal for year 2020, but the calculations were based on the closest available years, which in theory can be different for each material and type (crop o producer). Let's say cocoa impact calculations are based on 2010 producer h3data, 2009 crop h3data, and cocoa impact calculations are based on 2005 producer h3data and 2007 crop h3data. So we need to inform user about all those years for each requested material and type. But again final decision shall be made by FE / Science